src/audio/dai-zephyr.c

```diff
@@ -1054,7 +1054,12 @@ static int dai_comp_trigger_internal(struct comp_dev *dev, int cmd)
 		dai_trigger_op(dd->dai, cmd, dev->direction);
 #else
 		dai_trigger_op(dd->dai, cmd, dev->direction);
-		ret = dma_stop(dd->chan->dma->z_dev, dd->chan->index);
+		if (dev->state == COMP_STATE_ACTIVE) {
+			ret = dma_stop(dd->chan->dma->z_dev, dd->chan->index);
```

Comment:
I guess we have to do this now. I'm a bit worried about coupling device state and DMA state this way. This opens a problem if some state is set to not ACTIVE in some other piece of code. Then dma_stop() is never called.  Probably this is ok now, but it's actually pretty hard to verify this is the case by reading current dai-zephyr.c code.

OTOH, it seems there are no calls to get dma channel status with the public API. Right @teburd @juimonen ? dma_get_status() does not seem fit to gate calls to dma_stop().

Reply:
This will work but the best way would be to balance this in zephyr driver, in gpdma we already have channel internal state so calling dma_stop several times will not result in unbalanced pm state.

```
	/* Validate the channel state */
	if (chan_data->state != DW_DMA_ACTIVE &&
	    chan_data->state != DW_DMA_SUSPENDED) {
		ret = -EINVAL;
		goto out;
	}
```
but there is no such internal state check in HDA zephyr driver so we need to take care of this in SOF (this PR) **or** track device state in Zephyr (HDA driver)

Reply:
@abonislawski But as SOF code should work with any Zephyr driver, just fixing HDA is not enough, we'd need to have this as a guarantee at Zephyr dma.h level.

Reply:
at dma.h level we can only expose current state